### PR TITLE
Upstream animation-effect-timing-easing.html from Blink

### PR DESCRIPTION
**Pull request by alancutter**
Import https://chromium.googlesource.com/chromium/src/+/f9671263ba21966c21be2d695bc4d92f86db9ed2/third_party/WebKit/LayoutTests/web-animations-api/animation-effect-timing-easing.html

### CODE REVIEW

File: web-animations/interfaces/AnimationEffectTiming/easing.html [hunk 1]
```diff
@@ -40,17 +40,49 @@
...
+    '',
+    'test',
+    'initial',
+    'inherit',
+    'unset',
+    'steps(3, nowhere)',
+    'steps(-3, end)',
+    'cubic-bezier(0.1, 0, 4, 0.4)',
+    'function (a){return a}',
+    'function (x){return x}',
```
Comment (suzyh) [inline web-animations/interfaces/AnimationEffectTiming/easing.html:54]: I don't think we need to keep both linear cases for this test.

File: web-animations/interfaces/AnimationEffectTiming/easing.html [hunk 2]
```diff
@@ -40,17 +40,49 @@
...
+    'function(x, y){return 0.3}',
+    '7',
```
Comment (suzyh) [inline web-animations/interfaces/AnimationEffectTiming/easing.html]: Nit: I think this one should be moved further up the list. It seems a bit strange having it at the bottom after all the function-def and function-call items.

File: web-animations/interfaces/AnimationEffectTiming/easing.html [hunk 3]
```diff
@@ -40,17 +40,49 @@
...
-  assert_throws({ name: 'TypeError' },
-                function() {
-                  anim.effect.timing.easing = '';
-                });
-  assert_throws({ name: 'TypeError' },
-                function() {
-                  anim.effect.timing.easing = 'test';
-                });
-}, 'Test invalid easing value');
+  var anim = div.animate({});
```
Comment (suzyh) [inline web-animations/interfaces/AnimationEffectTiming/easing.html]: Do we need the div temporary? Maybe just have
var anim = createDiv(t).animate({});

Also, does it need to have keyframes, or could it be animate(null)?

File: web-animations/interfaces/AnimationEffectTiming/easing.html [hunk 4]
```diff
@@ -40,17 +40,49 @@
...
+    'steps(3, end)',
+    'cubic-bezier(0.1, 5, 0.23, 0)',
+  ];
+  var div = createDiv(t);
+  var anim = div.animate({});
+  for (var easing of roundTripEasings) {
+    anim.effect.timing.easing = easing;
+    assert_equals(anim.effect.timing.easing, easing);
+  }
+}, 'Test easing round trip equality');
```
Comment (suzyh) [inline web-animations/interfaces/AnimationEffectTiming/easing.html]: I think this test description is a little unclear. Maybe something like "Test that valid values for easing are returned unchanged"?

File: web-animations/interfaces/AnimationEffectTiming/easing.html [hunk 5]
```diff
@@ -40,17 +40,49 @@
...
+  var anim = div.animate({});
+  for (var invalidEasing of invalidEasings) {
+    assert_throws({ name: 'TypeError' }, function() {
+      anim.effect.timing.easing = invalidEasing;
+    }, invalidEasing);
+  }
+}, 'Test invalid easing values');
+
+test(function(t) {
+  var roundTripEasings = [
```
Comment (suzyh) [inline web-animations/interfaces/AnimationEffectTiming/easing.html:66]: What about the normalization tests from the imported file, e.g. "steps(3)" returning as "steps(3, end)"?

File: web-animations/interfaces/AnimationEffectTiming/easing.html (hunk 1 above)
Comment (alancutter) [inline web-animations/interfaces/AnimationEffectTiming/easing.html:54]: Firefox Nightly currently fails those cases, I think they're worth having.

File: web-animations/interfaces/AnimationEffectTiming/easing.html (hunk 5 above)
Comment (alancutter) [inline web-animations/interfaces/AnimationEffectTiming/easing.html:66]: Firefox fails this, I don't know if it's required by spec.

File: web-animations/interfaces/AnimationEffectTiming/easing.html (hunk 4 above)
Comment (alancutter) [inline web-animations/interfaces/AnimationEffectTiming/easing.html]: Done.

File: web-animations/interfaces/AnimationEffectTiming/easing.html (hunk 3 above)
Comment (alancutter) [inline web-animations/interfaces/AnimationEffectTiming/easing.html]: Done.

File: web-animations/interfaces/AnimationEffectTiming/easing.html (hunk 2 above)
Comment (alancutter) [inline web-animations/interfaces/AnimationEffectTiming/easing.html]: Done.

File: web-animations/interfaces/AnimationEffectTiming/easing.html (hunk 5 above)
Comment (suzyh) [inline web-animations/interfaces/AnimationEffectTiming/easing.html:66]: I think we should include it at least to make sure it gets brought up when we send these to Brian for review. It was discussed on www-style@w3.org (subject: "Normalization of steps()") but I don't think it was resolved yet.

Comment (alancutter) [inline web-animations/interfaces/AnimationEffectTiming/easing.html:66]: I'd rather bring it up in the PR comments than risk accidentally checking in a test for something undecided.

**Comment by suzyh** [comment]
lgtm

**Pull request merged**